### PR TITLE
Created a "read only administrator" authorization role 

### PR DESCRIPTION
**Pull request by okanji**
I have added support for a new user role **read_only_administrator** that has the permissions of a regular user and only the read permissions of an administrator. A **read_only_administrator** can not approve or reject a project but can view its information. It includes changes to the `authorization` script and authorization checks in various controller functions that require admin authorization as well as corresponding tests and changes to the web ui. For more background, see ticket #555 

##

These changes will allow 7-7000 agents to be able to view administrator data without being able to reject and review projects. 

##

Currently, **administrator** is the only role assigned to a human user. The **administrator** is associated with the`EditAll` authorization flag. A regular user has no associated authorization flags. 

I have added an additional flag `ReadAll` and added it to the **administrator** role in addition to the existing `EditAll` flag. The **read_only_administrator** is also assigned the `ReadAll` flag but not `EditAll`. The  **read_only_administrator** is also assigned the `ProvisionOnTestCluster` flag which is also assigned to the **administrator** role as it authorizes some fetch request functions but does not authorize any update functions (and so it does not give the **read_only_administrator** any write access).

##

Tests have been added for corresponding functions that use the newly created `ReadAll` authorization flag.

## 

I have disabled the review button in the web application for a **read_only_administrator**. I tried to delete the whole _response_ cell in the table but doing this caused the link in the name cell to become disabled. I am not sure why this is the case or how the two are connected and I chose not to dig deeper into this issue as it is purely an aesthetic issue and I feel that I have spent enough time on this PR already, but I can look further into it if needed.

### CODE REVIEW

File: web/src/views/Dashboard.tsx [hunk 1]
```diff
@@ -74,9 +74,10 @@ const Dashboard: React.FC = () => {
     verifyProjects();
   }, 1000 * 30);
 
+
   return (
     <>
-      {userRoles.includes('administrator') && <ProjectRequests profileDetails={profileDetails} />}
```
Comment (w8896699) [inline web/src/views/Dashboard.tsx:79]: I am not sure if we want to let them see all those requests, we can check this with Olena!

File: web/src/components/dashboard/ProjectRequests.tsx [hunk 2]
```diff
@@ -74,20 +74,22 @@ const ProjectRequests: React.FC<any> = (props) => {
...
-          </Box>
-        ),
+        Cell: ({ row: { values } }: any) =>
+          isAdminUser && (
+            <Box>
+              <button
+                type="button"
+                value={values.profileId}
+                onClick={() => {
+                  console.log('Click');
```
Comment (w8896699) [inline web/src/components/dashboard/ProjectRequests.tsx:84]: please remove console.log( I suggest we can add no console rule later in our eslint rule)

**Review by w8896699 — CHANGES_REQUESTED** [review]
Looks good to me, just a little change.

File: web/src/components/dashboard/ProjectRequests.tsx (hunk 2 above)
Comment (okanji) [reply web/src/components/dashboard/ProjectRequests.tsx]: Yes thank you for catching that!

**Comment by okanji** [comment]
@w8896699 re your comment about not being sure if the read only admin should be able to see all the requests, I am under the understanding that the read only admin should be able to see exactly what a regular admin sees. If you can be more specific in what you think that the read only admin I will relay that to Olena to get some direction

**Review by w8896699 — APPROVED** [review]
I agree with your comment above. Looks good to me now! I can show you how our pipeline works when you want to merge this pr.

**Pull request merged**